File: python/substr_count.py
import pprint
import unittest

o = [] # output
s0 = ""
s1 = "a"
s2 = "\n"
s3 = "a\nb"
s4 = "\nb"

def title(v):
    o.append("\n\n\n\n" + v + "\n")

def observation(v):
    o.append("\n\n" + v)

def var_export(v):
    o.append("\n" + pprint.pformat(v))



# wsgi header
def application(environ, start_response):
    status = '200 OK'

    o.append("<pre>")

    title('count empty string')
    res0 = s0.count("\n")
    var_export(res0)

    title('count empty string')
    res1 = s1.count("\n")
    var_export(res1)

    title('count empty string')
    res2 = s2.count("\n")
    var_export(res2)


    title('count empty string')
    res3 = s3.count("\n")
    var_export(res3)


    title('count empty string')
    res4 = s4.count("\n")
    var_export(res4)

    o.append("</pre>")

# wsgi footer
    response_headers = [('Content-type', 'text/html')]
    start_response(status, response_headers)
    return o

File: php/substr_count.php
<?php

function title($v)
{
    echo "\n\n\n\n$v\n\n";
}

function observation($v)
{
    echo "\n$v";
}

$s0 = "";
$s1 = "a";
$s2 = "a\n";
$s3 = "a\nb";
$s4 = "\nb";



echo "<pre>";

title('count empty string');
$res0 = substr_count($s0, "\n");
var_export($res0);
observation("");

title('count no occurrences');
$res1 = substr_count($s1, "\n");
var_export($res1);
observation("");

title('count as sufix');
$res2 = substr_count($s2, "\n");
var_export($res2);
observation("");

title('count as infix');
$res3 = substr_count($s3, "\n");
var_export($res3);
observation("");

title('count as prefix');
$res4 = substr_count($s4, "\n");
var_export($res4);
observation("");

echo "</pre>";

?>
